CP-51325: Re-add and update patches

ElijahSadorra opened the pull request:
Whilst writing CoP testing, encountered a bug which needed these previous patches. Have now updated the patches to be compatible with the new version of the library.

kc284 commented on patches/patch-sharpziplib-CA63358-TarBuffer.patch:
```diff
@@ -0,0 +1,126 @@
+
+--- a/SharpZipLib-1.3.3/src/ICSharpCode.SharpZipLib/Tar/TarBuffer.cs	2011-06-17 23:09:32.000000000 +0100
++++ b/SharpZipLib-1.3.3/src/ICSharpCode.SharpZipLib/Tar/TarBuffer.cs	2011-08-01 20:14:18.000000000 +0100
+@@ -286,55 +286,6 @@ namespace ICSharpCode.SharpZipLib.Tar
```
How did we end up with code lines in the chunk headers? (Several more instances of it below).

ElijahSadorra replied:
I'm assuming you mean the "namespace ICSharpCode.SharpZipLib.Tar"? If that's the case, it was auto generated by the git diff command. I tested with and without "namespace ICSharpCode.SharpZipLib.Tar", both are identical so happy to keep it so it is similar for future patches and also happy to remove it to keep the old convention.

kc284 replied:
As long as the patch applies successfully it doesn't matter too much, but I think it is cleaner without it.

ElijahSadorra replied:
Ok cool, yes both apply cleanly so will just remove it.

kc284 commented on patches/patch-sharpziplib-CA63358-TarBuffer.patch:
```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Cloud Software Group, Inc.
```
Minor: I suggest keeping the exact name of the original files (i.e. `patch-sharpziplib-ca-63358-TarBuffer.diff`) so it's easier to diff them.

ElijahSadorra commented:
Changed name and removed unnecessary use of namespace in the patches. When applying patches using `./build`, all patches was applied cleanly and produced the same `.dll`.
